network-config-analyzer/IstioSidecarYamlParser.py

```diff
@@ -0,0 +1,196 @@
+                if '*' not in dns_name:  # * means all services in namespace, all already in host_peers
+                    host_peers &= self._get_peers_from_host_dns_name(dns_name)
+            if special_case_host:
+                special_res_peers |= host_peers
+            else:
+                res_peers |= host_peers
+
+        return IstioSidecarRule(res_peers, special_res_peers)
+
+    def _check_and_save_sidecar_if_top_priority(self, curr_sidecar):
+        """
+        check if current sidecar is top priority for its namespace or workloads.
+        if the sidecar is selector less, and is the first default sidecar for current namespace, save it
+        otherwise, save it for any peer if this is the first sidecar selecting it
+        :param IstioSidecar curr_sidecar: the sidecar parsed in self
+        a warning message will be printed if current sidecar is not the first for its relevant object,
+        indicating that this sidecar will be ignored in the sidecar's connections
+        """
+        if curr_sidecar.default_sidecar:
+            if self.namespace.prior_default_sidecar:  # this sidecar is not first one
+                self.warning(f'Namespace "{str(self.namespace)}" already has a Sidecar configuration '
+                             f'without any workloadSelector. '
+                             f'Connections in sidecar: "{curr_sidecar.full_name()}" will be ignored')
+                return
+            self.namespace.prior_default_sidecar = curr_sidecar
+            return
+
+        for peer in curr_sidecar.selected_peers:
+            if peer.prior_sidecar:  # this sidecar is not first one
+                self.warning(f'Peer "{peer.full_name()}" already has a Sidecar configuration selecting it. Sidecar: '
+                             f'"{curr_sidecar.full_name()}" will not be considered as connections for this workload')
+                return
+            peer.prior_sidecar = curr_sidecar
+
+    def parse_policy(self):
+        """
+        Parses the input object to create a IstioSidecar object
+        :return: a IstioSidecar object with proper PeerSets
+        :rtype: IstioSidecar
+        """
+        policy_name = self.parse_generic_istio_policy_fields('Sidecar', 'networking.istio.io/v1beta1')
+        if policy_name is None:
+            return None  # not relevant to build this policy
+
+        res_policy = IstioSidecar(policy_name, self.namespace)
+        res_policy.policy_kind = NetworkPolicy.PolicyType.IstioSidecar
+
+        if 'spec' not in self.policy or self.policy['spec'] is None:
+            self.syntax_error('spec is missing or null in Sidecar ' + res_policy.full_name())
+
+        sidecar_spec = self.policy['spec']
+        # currently, supported fields in spec are workloadSelector and egress
+        allowed_spec_keys = {'workloadSelector': [0, dict], 'ingress': [3, list], 'egress': [0, list],
+                             'outboundTrafficPolicy': [2, str]}
```

Comment:
Just for this version, since current version handles only internal services ...so this field is not supported yet. 
may change the code to 3 and skip using it for now ( in the next sidecar PR , this will be supported with adding support of external services)

Reply:
ok, I think it is better to skip using code 3 for now.